cmd2/cmd2.py

```diff
@@ -379,21 +373,20 @@ def __init__(self, completekey: str = 'tab', stdin=None, stdout=None, persistent
         self.aliases = dict()
         self.macros = dict()
 
-        self._finalize_app_parameters()
-
         self.initial_stdout = sys.stdout
         self.history = History()
         self.pystate = {}
         self.py_history = []
         self.pyscript_name = 'app'
-        self.keywords = self.reserved_words + self.get_all_commands()
-        self.statement_parser = StatementParser(
-            allow_redirection=self.allow_redirection,
-            terminators=self.terminators,
-            multiline_commands=self.multiline_commands,
-            aliases=self.aliases,
-            shortcuts=self.shortcuts,
-        )
+
+        if shortcuts is None:
+            shortcuts = self.DEFAULT_SHORTCUTS
+        shortcuts = sorted(shortcuts.items(), reverse=True)
+        self.statement_parser = StatementParser(allow_redirection=allow_redirection,
+                                                terminators=terminators,
+                                                multiline_commands=multiline_commands,
+                                                aliases=self.aliases,
```

Comment:
I think a method to retrieve aliases should be passed into `self.statement_parser`. Just in case someone ever recreates `self.aliases` after the creation of `self.statement_parser`.

Reply:
I was debating doing one of two things for this:
1. Pass in a callback function to the ``StatementParser`` that it uses to get the latest aliases
2. Create a read/write property in ``cmd2.Cmd`` that is used to access the master copy of aliases within ``StatementParser`` and not maintain its own copy

Do you think that one of this approaches has architectural advantages over the other?

Reply:
Normally I'd say having the parser call back up to its owner isn't the best approach. But we do that in other places in `cmd2`. However, since the parser already owns the other settings, maybe just let it own aliases, too.

Reply:
Aliases are now stored solely in the ``StatementParser`` instance and I added a read-only property to ``cmd2.Cmd`` to get the aliases.